CurrencyConverterApplication/ViewModel/ProductViewItem.cs

```diff
@@ -0,0 +1,51 @@
+﻿using CurrencyConverterApplication.Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CurrencyConverterApplication.ViewModel
+{
+    public  class ProductViewItem:ViewModelBase
+    {
+        private readonly Product _model;
+        public ProductViewItem(Product model)
+        {
+            _model = model;
+        }
+
+       
+
+        public string? ProductName
+        {
+            get => _model.ProductName;
+            set
+            {
+                _model.ProductName = value;
+                RaisePropertychanged();                
```

Comment:
You should only raise the event if the value has actually been changed.

Reply:
The RaisePropertyChanged() method is actually a calling of the INotifyPropertyChanged interface, which Invokes an event only when the property has changed its value. 
thanks